examples/rust/examples/fibonacci.rs

```diff
@@ -0,0 +1,22 @@
+#![no_main]
+
+// In this example built-in field type is used.
+type PallasBase = __zkllvm_field_pallas_base;
```

Comment:
There is no reason for that. I just copied from _builtins example
Is `arkworks` our next destination?
Should I rewrite it for arkworks?

Reply:
We aim to use `arkworks` in any user-side code and examples are exactly the case. 

The reason for that is to ensure that code makes sense for native target: you can compile and run this code if you want. And with raw built-in types this makes no sense - they simply do not exist in any native code. 

So these field and curve built-ins are sort of special markers for compiler and assigner, not a real user types. 

It would be perfect if you could rewrite this with `arkworks` - this requires changing only one line :)

Reply:
reworked in cee61dd